### PR TITLE
support generic TF models

### PR DESCRIPTION
**Pull request by imakunin**

### CODE REVIEW

File: serialzy/base.py [hunk 1]
```diff
@@ -77,5 +77,8 @@ def resolve(self, schema: Schema) -> Type:
         if module == 'builtins' and name == 'NoneType':
             return type(None)
 
-        typ = getattr(importlib.import_module(module), name)
-        return cast(type, typ)
+        if name is not None:
+            typ = getattr(importlib.import_module(module), name)
+            return cast(type, typ)
```
Comment (vhaldemar) [inline serialzy/base.py:82]: maybe `cast(Type` instead of Type?

File: serialzy/serializers/tensorflow.py [hunk 2]
```diff
@@ -43,6 +45,20 @@ class TensorflowPureSerializer(ModelBaseSerializer):
     def __init__(self):
         super().__init__("tensorflow", __name__)
 
+    def schema(self, typ: Type) -> Schema:
+        if self._is_generic_user_object(typ):
+            return Schema(
+                self.data_format(),
```
Comment (vhaldemar) [inline serialzy/serializers/tensorflow.py:51]: please, use named arguments here

File: serialzy/serializers/tensorflow.py [hunk 3]
```diff
@@ -43,6 +45,20 @@ class TensorflowPureSerializer(ModelBaseSerializer):
...
+                self.SCHEMA_FORMAT,
+                json.dumps({
+                    "module": self.module,
+                    "name": None
+                }),
+                self.meta(),
+            )
+        else:
```
Comment (vhaldemar) [inline serialzy/serializers/tensorflow.py:59]: else is unnecesary here because of upper return

File: serialzy/serializers/tensorflow.py [hunk 4]
```diff
@@ -62,4 +78,14 @@ def _deserialize(self, source: BinaryIO, schema_type: Type, user_type: Optional[
 
     def _types_filter(self, typ: Type):
         import tensorflow as tf  # type: ignore
-        return typ in [tf.train.Checkpoint] or (inspect.isclass(typ) and issubclass(typ, tf.Module))
+        if typ in [tf.train.Checkpoint]:
```
Comment (vhaldemar) [inline serialzy/serializers/tensorflow.py:81]:
```
return (
    typ in [tf.train.Checkpoint] or 
    inspect.isclass(typ) and issubclass(typ, tf.Module) or
    self._is_generic_user_object(typ)
)
```

File: serialzy/serializers/tensorflow.py [hunk 5]
```diff
@@ -62,4 +78,14 @@ def _deserialize(self, source: BinaryIO, schema_type: Type, user_type: Optional[
...
+            return True
+        if inspect.isclass(typ) and issubclass(typ, tf.Module):
+            return True
+        return self._is_generic_user_object(typ)
+
+    @staticmethod
+    def _is_generic_user_object(typ: Type) -> bool:
+        return inspect.isclass(type)\
```
Comment (vhaldemar) [inline serialzy/serializers/tensorflow.py:89]: use 
````
return (
   foo and
   bar
)
```
intendations please

File: serialzy/serializers/tensorflow.py [hunk 6]
```diff
@@ -62,4 +78,14 @@ def _deserialize(self, source: BinaryIO, schema_type: Type, user_type: Optional[
...
+               and typ.__module__ == "tensorflow.python.saved_model.load"\
```
Comment (vhaldemar) [inline serialzy/serializers/tensorflow.py:90]: looks nasty

Comment (vhaldemar) [reply serialzy/serializers/tensorflow.py:90]: also, im not 100% sure that any class object will have `__module__` attribute, maybe you should use `getattr(typ, '__module__', '') == "tensorflow.python.saved_model.load"` for better safety instead

File: serialzy/base.py (hunk 1 above)
Comment (imakunin) [reply serialzy/base.py:82]: извињавам се, али нисам разумео

File: serialzy/serializers/tensorflow.py [hunk 7]
```diff
@@ -62,4 +77,16 @@ def _deserialize(self, source: BinaryIO, schema_type: Type, user_type: Optional[
...
+            typ in [tf.train.Checkpoint] or
+            (inspect.isclass(typ) and issubclass(typ, tf.Module)) or
+            self._is_generic_user_object(typ)
+        )
+
+    @staticmethod
+    def _is_generic_user_object(typ: Type) -> bool:
+        return (
+            inspect.isclass(typ) and
+            getattr(typ, "__module__") == "tensorflow.python.saved_model.load" and
```
Comment (vhaldemar) [inline serialzy/serializers/tensorflow.py:90]: eh, it makes sense to change dot to getattr if you passing third "default" argument to getattr

**Pull request merged**